src/QuantumSdk/DefaultItems/DefaultItems.targets

```diff
@@ -59,10 +59,13 @@
   <PropertyGroup>
     <DefaultEntryPointGeneration>dotnet "$(MSBuildThisFileDirectory)../tools/utils/Microsoft.Quantum.Sdk.DefaultEntryPoint.Generation.dll"</DefaultEntryPointGeneration>
     <!-- true/false optional properties -->
-    <QirGeneration Condition="'$(QirOutputPath)' != ''">true</QirGeneration>
+    <QirGeneration Condition="'$(QirOutputPath)' != '' And '$(QirGeneration)' == 'default'">true</QirGeneration>
```

Comment:
I am a bit confused why that works; aren't you missing an update to the DefaultItems.props.template to set the value to default initially? Also, did you check that all three *Generation properties are used within the Sdk only after they are set to either true or false here?